src/runtime/contrib/dnnl/dnnl_json_runtime.cc

```diff
@@ -166,100 +218,88 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
+    dnnl::memory::dim channels = std::stoi(node.GetAttr<std::vector<std::string>>("channels")[0]);
     std::vector<std::string> str_strides = node.GetAttr<std::vector<std::string>>("strides");
     std::vector<std::string> str_dilates = node.GetAttr<std::vector<std::string>>("dilation");
     std::vector<std::string> str_padding = node.GetAttr<std::vector<std::string>>("padding");
+    std::vector<std::string> str_padding_l(str_padding.begin(),
+                                           str_padding.begin() + str_padding.size() / 2);
+    std::vector<std::string> str_padding_r(str_padding.end() - str_padding.size() / 2,
+                                           str_padding.end());
     dnnl::memory::dim groups = std::stoi(node.GetAttr<std::vector<std::string>>("groups")[0]);
+    std::string data_layout = node.GetAttr<std::vector<std::string>>("data_layout")[0];
+    std::string kernel_layout = node.GetAttr<std::vector<std::string>>("kernel_layout")[0];
 
-    dnnl::memory::dim N = input_shape[0],        // batch size
-        IC = input_shape[1],                     // input channels
-        IH = input_shape[2],                     // input height
-        IW = input_shape[3],                     // input width
-        OC = weight_shape[0],                    // output channels
-        KH = weight_shape[2],                    // weight height
-        KW = weight_shape[3],                    // weight width
-        PW_L = std::stoi(str_padding[1]),        // width padding: left
-        PW_R = std::stoi(str_padding[3]),        // width padding: right
-        PH_L = std::stoi(str_padding[0]),        // height padding: top
-        PH_R = std::stoi(str_padding[2]),        // height padding: bottom
-        SH = std::stoi(str_strides[0]),          // height-wise stride
-        SW = std::stoi(str_strides[1]),          // weight-wise stride
-        DH = std::stoi(str_dilates[0]) - 1,      // height-wise dilate
-        DW = std::stoi(str_dilates[1]) - 1,      // weight-wise dilate
-        DKH = 1 + (KH - 1) * (DH + 1),           // dilated weight height
-        DKW = 1 + (KW - 1) * (DW + 1),           // dilated weight width
-        OH = (IH - DKH + PH_L + PH_R) / SH + 1,  // output height
-        OW = (IW - DKW + PW_L + PW_R) / SW + 1;  // output width
+    // Check layout.
+    if (layout_dict.find(data_layout) == layout_dict.end() ||
+        layout_dict.find(kernel_layout) == layout_dict.end()) {
+      LOG(FATAL) << "Unsupported layout for conv: " << data_layout << " " << kernel_layout;
+    }
 
     // Memory shapes.
-    dnnl::memory::dims src_dims = {N, IC, IH, IW};
-    dnnl::memory::dims weights_dims = {OC, IC, KH, KW};
+    dnnl::memory::dims src_dims = input_shape;       // {N, IC, ID, IH, IW}
+    dnnl::memory::dims weights_dims = weight_shape;  // {OC, IC, KD, KH, KW}
     if (groups > 1) {
-      weights_dims = {groups, 1, IC / groups, KH, KW};
+      weights_dims = {groups, 1, input_shape[1] / groups};
```

Comment:
What's the value for `oc` dimension?

Reply:
replace `1` with `channels / groups`, which equals to `OC`.